src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -2,41 +2,64 @@ import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Course, CourseScoreCalculationService, CourseService } from 'app/entities/course';
 import { ActivatedRoute } from '@angular/router';
 import { Subscription } from 'rxjs/Subscription';
-import { LangChangeEvent, TranslateService } from '@ngx-translate/core';
+import { TranslateService } from '@ngx-translate/core';
 import { HttpResponse } from '@angular/common/http';
 import * as moment from 'moment';
 import { Exercise, ExerciseService } from 'app/entities/exercise';
+import { AccountService } from 'app/core';
+import { sum } from 'lodash';
+
+enum ExerciseFilter {
+    OVERDUE = 'OVERDUE',
+    NEEDS_WORK = 'NEEDS_WORK',
+}
+
+enum ExerciseSortingOrder {
+    DUE_DATE_ASC = 1,
+    DUE_DATE_DESC = -1,
+}
+
+enum SortFilterStorageKey {
+    FILTER = 'artemis.course.exercises.filter',
+    ORDER = 'artemis.course.exercises.order',
+}
 
 @Component({
     selector: 'jhi-course-exercises',
     templateUrl: './course-exercises.component.html',
     styleUrls: ['../course-overview.scss'],
 })
 export class CourseExercisesComponent implements OnInit, OnDestroy {
-    public readonly DUE_DATE_ASC = 1;
-    public readonly DUE_DATE_DESC = -1;
     private courseId: number;
     private paramSubscription: Subscription;
     private translateSubscription: Subscription;
     public course: Course | null;
     public weeklyIndexKeys: string[];
     public weeklyExercisesGrouped: object;
-
     public upcomingExercises: Exercise[];
-
     public exerciseCountMap: Map<string, number>;
 
+    readonly ASC = ExerciseSortingOrder.DUE_DATE_ASC;
+    readonly DESC = ExerciseSortingOrder.DUE_DATE_DESC;
+    sortingOrder: ExerciseSortingOrder;
+    activeFilters: Set<ExerciseFilter>;
+
     constructor(
         private courseService: CourseService,
         private courseCalculationService: CourseScoreCalculationService,
         private courseServer: CourseService,
         private translateService: TranslateService,
         private exerciseService: ExerciseService,
+        private accountService: AccountService,
         private route: ActivatedRoute,
     ) {}
 
     ngOnInit() {
         this.exerciseCountMap = new Map<string, number>();
+        const filters = localStorage.getItem(SortFilterStorageKey.FILTER);
```

Comment:
I think we should move interactions with the `localStorage` API into it's  own service. (We could do thin in a follow up)